jwst/tweakreg/tweakreg_step.py

```diff
@@ -277,34 +283,36 @@ def process(self, input):
-                                       ref_wcsinfo=ref_image.meta.wcsinfo.instance,
-                                       epoch=Time(ref_image.meta.observation.date).decimalyear,
-                                       abs_minobj=self.abs_minobj,
-                                       abs_fitgeometry=self.abs_fitgeometry,
-                                       abs_nclip=self.abs_nclip,
-                                       abs_sigma=self.abs_sigma,
-                                       abs_searchrad=self.abs_searchrad,
-                                       abs_use2dhist=self.abs_use2dhist,
-                                       abs_separation=self.abs_separation,
-                                       abs_tolerance=self.abs_tolerance,
-                                       save_abs_catalog=self.save_abs_catalog,
-                                       abs_catalog_output_dir=self.output_dir,
-                                            )
-
-            except twk.TweakregError as e:
-                self.log.warning(str(e))
-                for model in images:
-                    model.meta.cal_step.tweakreg = "SKIPPED"
-                return images 
-
-        if local_align_failed and not align_to_abs_refcat:    
-            for model in images:
-                record_step_status(model, "tweakreg", success=False)
+            with images:
+                try:
+                    ref_image = images.borrow(0)
+                    correctors = \
+                        twk.absolute_align(correctors, self.abs_refcat,
+                                        ref_wcs=ref_image.meta.wcs,
+                                        ref_wcsinfo=ref_image.meta.wcsinfo.instance,
+                                        epoch=Time(ref_image.meta.observation.date).decimalyear,
+                                        abs_minobj=self.abs_minobj,
+                                        abs_fitgeometry=self.abs_fitgeometry,
+                                        abs_nclip=self.abs_nclip,
+                                        abs_sigma=self.abs_sigma,
+                                        abs_searchrad=self.abs_searchrad,
+                                        abs_use2dhist=self.abs_use2dhist,
+                                        abs_separation=self.abs_separation,
+                                        abs_tolerance=self.abs_tolerance,
+                                        save_abs_catalog=self.save_abs_catalog,
+                                        abs_catalog_output_dir=self.output_dir,
+                                                )
+                    images.shelve(ref_image, 0, modify=False)
+                    del ref_image
+
+                except twk.TweakregError as e:
+                    self.log.warning(str(e))
+                    images.shelve(ref_image, 0, modify=False)
+                    record_step_status(images, "tweakreg", success=False)
+                    return images
```

Comment:
```suggestion
                except twk.TweakregError as e:
                    self.log.warning(str(e))
                    record_step_status(images, "tweakreg", success=False)
                    return images
                finally:
                    images.shelve(ref_image, 0, modify=False)
                    del ref_image
```
```suggestion
                    images.shelve(ref_image, 0, modify=False)
                    del ref_image

                except twk.TweakregError as e:
                    self.log.warning(str(e))
                    images.shelve(ref_image, 0, modify=False)
                    record_step_status(images, "tweakreg", success=False)
                    return images
```

Reply:
This doesn't work because `record_step_status` expects a closed library.  Let me know if you have a suggestion for how to refactor these lines, but since only one model is open it makes sense to me that you would just shelve it, then re-open it as part of looping over all models in record_sep_status.